File: apps/mybookkeeper/frontend/src/shared/types/welcome-manual/welcome-manual-room-response.ts
/**
 * Mirrors backend `WelcomeManualRoomResponse` — one lettable room within a
 * by-the-room manual. Rooms hold no content themselves; sections point at them
 * via `room_id`, and a section with `room_id === null` is shared by every room.
 */
export interface WelcomeManualRoomResponse {
  id: string;
  manual_id: string;
  name: string;
  display_order: number;
  created_at: string;
  updated_at: string;
}

File: apps/mybookkeeper/backend/app/schemas/welcome_manuals/welcome_manual_room_response.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class WelcomeManualRoomResponse(BaseModel):
    """One room within a by-the-room welcome manual.

    Rooms don't carry content themselves — sections point at them. A section
    with ``room_id = None`` is shared by every room; one carrying this room's
    id appears only in this room's guide.
    """

    id: uuid.UUID
    manual_id: uuid.UUID
    name: str
    display_order: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
